lib/pack/universal-generator/test/assertGenerator.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    guid,
+    HandlerContext,
+    Maker,
+    MessageClient,
+    Project,
+    ProjectOperationCredentials,
+    RepoId,
+} from "@atomist/automation-client";
+import { successOn } from "@atomist/automation-client/lib/action/ActionResult";
+import { HandleCommand } from "@atomist/automation-client/lib/HandleCommand";
+import { HandleEvent } from "@atomist/automation-client/lib/HandleEvent";
+import { BuildableAutomationServer } from "@atomist/automation-client/lib/server/BuildableAutomationServer";
+import {
+    AbstractSoftwareDeliveryMachine,
+    GeneratorRegistration,
+    generatorRegistrationToCommand,
+    SoftwareDeliveryMachineConfiguration,
+} from "@atomist/sdm";
+import * as assert from "assert";
+import * as flatten from "flat";
+import * as _ from "lodash";
+import { defaultSoftwareDeliveryMachineConfiguration } from "../../../machine/defaultSoftwareDeliveryMachineConfiguration";
+import { toArray } from "../../../util/misc/array";
+import { invokeCommand } from "../../job/invokeCommand";
+import { universalGenerator } from "../generator";
+import { UniversalTransform } from "../generatorSupport";
+
+export interface AssertGeneratorResult {
+    id: RepoId;
+    project: Project;
+    parameters: Record<string, any>;
+}
+
+export async function assertUniversalGenerator(generatorUnderTest: GeneratorRegistration<any>,
+                                               transformsUnderTest: UniversalTransform<any> | Array<UniversalTransform<any>>,
+                                               initialParams: Record<string, any>,
+                                               promptForParams: Record<string, any> = {}): Promise<AssertGeneratorResult> {
+    try {
+        // Prepare the result of generator run
+        let project: Project = undefined as any;
+        let id: RepoId = undefined as any;
+        let params = undefined as any;
```

Comment:
This could just be:

```suggestion
        let project: Project;
        let id: RepoId;
        let params: any;
```

Reply:
So multiline comments are a thing, but not multiline suggestions in those comments.

Reply:
Oddly your suggestion didn't work in the aspect seed project where I developed this. But it does work in sdm-core which probably as more lax tsconfig settings.